### PR TITLE
Provide GatherND with original layout for inputs and output

### PR DESCRIPTION
**Pull request by rkazants**
Description: GatherND accepts inputs and output in NHWC layout in case TensorFlow so implemented transformation cares of it.

JIRA: 41818

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests
* [x]  Framework operation tests - N/A
* [x]  Transformation tests
* [x]  e2e model test with an update of ./tests/e2e_oss/utils/reshape_utils.py - N/A
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list
* [x]  Supported **public** models list - N/A
* [x]  New operations specification
* [x]  Guide on how to convert the **public** model - N/A
* [x]  User guide update - N/A

Signed-off-by: Roman Kazantsev <roman.kazantsev@intel.com>

### CODE REVIEW

File: model-optimizer/extensions/back/LayoutChangeForGatherND.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    Return original layout for inputs and output of GatherND operation
+    since the operation is designed for NHWC layout.
+    """
+    enabled = True
+    graph_condition = [lambda graph: graph.graph['fw'] == 'tf']
+
+    @staticmethod
+    def insert_transpose(graph: Graph, input_port: Port, before_input=True):
+        input_rank = len(input_port.data.get_shape())
+        if input_rank > 3 and before_input:
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/back/LayoutChangeForGatherND.py:36]: We can refactor the code significantly. Something like this:

```
input_rank = len(input_port.data.get_shape())
if input_rank > 3:
    if before_input:
         axis_order = np.concatenate((int64_array([0]),
                                         int64_array(list(range(2, input_rank))),
                                         int64_array([1])))
    else:
         axis_order = np.concatenate(
                (int64_array([0]),
                 int64_array([input_rank - 1]),
                 int64_array(list(range(1, input_rank - 1)))))
  transpose = create_op_with_const_inputs(graph, Transpose, {1: axis_order},  {'name': 'Transpose'})
  input_port.get_connection().insert_node(transpose)
  transpose['need_shape_inference'] = True
  transpose['override_output_shape'] = True
```

File: model-optimizer/extensions/back/LayoutChangeForGatherND.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            input_port.get_connection().set_destination(transpose.in_port(0))
+            input_port.connect(transpose.out_port(0))
+            transpose['need_shape_inference'] = True
+            transpose['override_output_shape'] = True
+        elif input_rank > 3:
+            axis_order = np.concatenate(
+                (int64_array([0]),
+                 int64_array([input_rank - 1]),
+                 int64_array(list(range(1, input_rank - 1)))))
+            transpose = create_op_with_const_inputs(graph, Transpose, {1: axis_order},
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/back/LayoutChangeForGatherND.py:51]: Please, use more specific node name based on the name of the `input_port.node.name`

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: model-optimizer/extensions/back/LayoutChangeForGatherND.py (hunk 1 above)
Comment (rkazants) [reply model-optimizer/extensions/back/LayoutChangeForGatherND.py]: Cool! Thanks for this update.

File: model-optimizer/extensions/back/LayoutChangeForGatherND.py (hunk 2 above)
Comment (rkazants) [reply model-optimizer/extensions/back/LayoutChangeForGatherND.py]: Done.

File: model-optimizer/extensions/back/LayoutChangeForGatherND.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+
+class LayoutChangeForGatherND(BackReplacementPattern):
+    """
+    Return original layout for inputs and output of GatherND operation
+    since the operation is designed for NHWC layout.
+    """
+    enabled = True
+    graph_condition = [lambda graph: graph.graph['fw'] == 'tf']
+
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/back/LayoutChangeForGatherND.py:32]: need `force_shape_inference = True`

Comment (rkazants) [reply model-optimizer/extensions/back/LayoutChangeForGatherND.py:33]: Fixed.

**Review by lazarevevgeny — APPROVED** [review]

**Comment by lazarevevgeny** [comment]
@rkazants, please, pick up changes from the recent master to pass OpenVINO-ONNX job

**Pull request merged**